src/graphql/account.rs

```diff
@@ -418,6 +418,49 @@ impl AccountMutation {
         update_jwt_expires_in(expires_in)?;
         Ok(time)
     }
+
+    /// Retrieves the user's language selection by username.
+    #[graphql(guard = "RoleGuard::new(super::Role::SystemAdministrator)
+        .or(RoleGuard::new(super::Role::SecurityAdministrator))
+        .or(RoleGuard::new(super::Role::SecurityManager))
+        .or(RoleGuard::new(super::Role::SecurityMonitor))")]
+    async fn language(&self, ctx: &Context<'_>, username: String) -> Result<Option<String>> {
+        let store = crate::graphql::get_store(ctx).await?;
+        let map = store.account_map();
+
+        if let Some(account) = map.get(&username).map_err(|e| e.to_string())? {
```

Comment:
I don't think `map_err` is necessary here. And, using `ok_or` would make the `if else` clause unnecessary.

Reply:
I rewrote a code that applied what you said by referring to other functions.